integrated_channels/integrated_channel/exporters/learner_data.py

```diff
@@ -80,6 +83,133 @@ def grade_audit(self):
         """
         return self.GRADE_AUDIT
 
+    def fetch_course_details(self, course_details, enrollment_course_id):
+        """
+        Placeholder
+        """
+        # Check if first run or if the cached course ID doesn't match the enrollment course ID
+        if course_details is None or course_details['course_id'] != enrollment_course_id:
```

Comment:
The fact that it looks like the EnterpriseCourseEnrollment table stores course_run_ids (which I see us querying off of in another file) and yet every part of the course api seems to say course_id even though it can use course_run....my brain. It's dying.

I was going to ask for consistently using `course_run_id` as we do later in this file, but since everything externally uses course_id as well it doesn't really help my confusion.

_this has been useless comments by Talia™️_

Reply:
Off the top of my head, I've seen mixtures of `course_id`, `courseID`, and `course_run_id` all pretty much used for the same thing.. and yet not. It's my understanding that most of this confusion came from the fact that the original integrated channels didn't know whether or not it was dealing with a course or a course run, so they had to compensate for both. With assessment level, I don't believe we will ever get an assessment level grade from a course (please please correct me if I'm wrong because then I have some more work to do 🙃 ), so there is an argument to making a switch.

Reply:
Yeah it's my understanding we wouldn't be able to have assessments without course runs. However, I think using course_run and having to say course_id==course_run in spots doesn't do much to alleviate the issue, which is why I said my comment was just useless lol.

I imagine we _could_ switch everything enrollment related, but not sure it's worth it at this point and not sure if we'll still hit spots where we equate a "course_id" with course run id anyways.

Reply:
in my eyes, defs not worth lol